app/src/main/java/com/epfl/neighborfood/neighborfoodandroid/models/Conversation.java

```diff
@@ -1,30 +1,67 @@
 package com.epfl.neighborfood.neighborfoodandroid.models;
 
-import com.google.android.gms.tasks.Task;
 
+import com.epfl.neighborfood.neighborfoodandroid.authentication.AuthenticatorFactory;
 import java.util.List;
+import java.util.Set;
 
-public class Conversation extends Model{
-    private final User chatter ;
+/**
+ * @author Mohamed Yassine Boukhari
+ *
+ */
+public class Conversation extends Model {
+    private final Set<User> users;
     private final List<Message> messages;
 
-
-    public Conversation(User chatter, List<Message> messages) {
-        this.chatter = chatter;
+    /**
+     * constructor for the conversation class
+     * @param users the set of users(2) in this conversation
+     * @param messages the initial list of messages to start
+     * the conversation with
+     */
+    public Conversation(Set<User> users, List<Message> messages) {
+        this.users = users;
         this.messages = messages;
     }
 
+    /**
+     * used to get the user with whom the current logged in
+     * user is chatting
+     * @return a User
+     */
     public User getChatter() {
+        User chatter = null;
+        for (User usr: users){
+            if(!usr.getId().equals(AuthenticatorFactory.getDependency().getCurrentUser().getId())){
+                chatter = usr;
```

Comment:
You should take into consideration the case when the authenticator returns a null value (when there's no logged in user). This is an edge case that souldn't be happening, but I think it's good practice to catch that to avoid runtime exceptions